pkg/labels/consul_applicator.go

```diff
@@ -124,6 +117,22 @@ func (c *consulApplicator) GetMatches(selector labels.Selector, labelType Type,
 	return res, nil
 }
 
+func (c *consulApplicator) ListLabels(labelType Type) ([]Labeled, error) {
```

Comment:
I'd like to see (at least in-memory) rate limiting in here someday. This is a heavy function to call.